From 6963bb29083910204f53924476f54f23c6578b86 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Update: xzl <xiangzelong@deepin.org>
Date: Fri, 29 Nov 2024 13:55:39 +0800
Subject: [PATCH 1/1] add more Gallium and Vulkan drivers for all architectures

- This is a pre-upstream patch, I will split this into multiple ones to
  justify each.
- Enable svga Gallium driver by default on ARM targets as VMware can run ARM
  guest machines.
- Enable etnaviv Gallium driver by default on all architectures to support
  older JEMO JM-series GPUs.
- The xe kernel driver is now available and possible to be utilised on all
  known architectures listed in meson.build. Enable iris Gallium drivers
  on all architectures by default.
- Enable more Vulkan drivers following the same spirit as above.
---
 meson.build | 25 +++++++++++++++++--------
 1 file changed, 17 insertions(+), 8 deletions(-)

diff --git a/meson.build b/meson.build
index 52b713c2..651a6e69 100644
--- a/meson.build
+++ b/meson.build
@@ -129,21 +129,21 @@ if gallium_drivers.contains('auto')
     if ['x86', 'x86_64'].contains(host_machine.cpu_family())
       gallium_drivers = [
         'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'svga', 'llvmpipe', 'softpipe',
-        'iris', 'crocus', 'i915', 'zink'
+        'iris', 'crocus', 'i915', 'zink', 'etnaviv"'
       ]
     elif ['arm', 'aarch64'].contains(host_machine.cpu_family())
       gallium_drivers = [
         'v3d', 'vc4', 'freedreno', 'etnaviv', 'nouveau', 'svga',
         'tegra', 'virgl', 'lima', 'panfrost', 'llvmpipe', 'softpipe', 'iris',
-        'zink'
+        'zink', 'r300', 'r600', 'radeonsi', 'svga', 'asiha'
       ]
     elif ['mips', 'mips64', 'ppc', 'ppc64', 'riscv32', 'riscv64'].contains(host_machine.cpu_family())
       gallium_drivers = [
-        'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'llvmpipe', 'softpipe', 'zink'
+        'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'llvmpipe', 'softpipe', 'zink', 'iris', 'etnaviv'
       ]
     elif ['loongarch64'].contains(host_machine.cpu_family())
       gallium_drivers = [
-        'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'etnaviv', 'llvmpipe', 'softpipe', 'zink'
+        'r300', 'r600', 'radeonsi', 'nouveau', 'virgl', 'etnaviv', 'llvmpipe', 'softpipe', 'zink', 'iris'
       ]
     else
       error('Unknown architecture @0@. Please pass -Dgallium-drivers to set driver options. Patches gladly accepted to fix this.'.format(
@@ -219,13 +219,22 @@ _vulkan_drivers = get_option('vulkan-drivers')
 if _vulkan_drivers.contains('auto')
   if system_has_kms_drm
     if host_machine.cpu_family().startswith('x86')
-      _vulkan_drivers = ['amd', 'intel', 'intel_hasvk', 'nouveau', 'swrast']
+      _vulkan_drivers = [
+        'amd', 'intel', 'intel_hasvk', 'nouveau', 'swrast', 'virtio'
+      ]
     elif ['arm', 'aarch64'].contains(host_machine.cpu_family())
-      _vulkan_drivers = ['swrast', 'intel']
+      _vulkan_drivers = [
+        'amd', 'broadcom', 'freedreno', 'intel', 'nouveau', 'panfrost',
+        'swrast', 'virtio'
+      ]
     elif ['mips', 'mips64', 'ppc', 'ppc64', 'riscv32', 'riscv64'].contains(host_machine.cpu_family())
-      _vulkan_drivers = ['amd', 'swrast']
+      _vulkan_drivers = [
+        'amd', 'intel', 'nouveau', 'swrast', 'virtio'
+      ]
     elif ['loongarch64'].contains(host_machine.cpu_family())
-      _vulkan_drivers = ['amd', 'swrast']
+      _vulkan_drivers = [
+        'amd', 'intel', 'nouveau', 'swrast', 'virtio'
+      ]
     else
       error('Unknown architecture @0@. Please pass -Dvulkan-drivers to set driver options. Patches gladly accepted to fix this.'.format(
             host_machine.cpu_family()))
-- 
2.45.2
